lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -743,6 +743,11 @@ static bool canonicalizeLogicalCstWithConcat(Operation *logicalOp,
   return true;
 }
 
+static bool canCombineOppositeCmpIntoConstant(ICmpOp lhs, ICmpOp rhs) {
```

Comment:
This is only valid for 'bin' flagged ops.  The bin flag marks the op as only caring about the binary truth-table (rather than the vhdl or verilog truth-tables)

Reply:
Thanks, hadn't thought about that. I've added a check for that. This wouldn't affect the `andOp` or `orOp` itself, correct?